src/main/frontend/components/cmdk.cljs

```diff
@@ -498,22 +498,23 @@
         create-whiteboard? (= :whiteboard (:source-create item))
         create-page? (= :page (:source-create item))
         class (when create-class? (get-class-from-input @!input))]
-    (cond
-      create-class? (page-handler/create! class
-                                          {:redirect? false
-                                           :create-first-block? false
-                                           :class? true})
-      create-whiteboard? (whiteboard-handler/create-new-whiteboard-and-redirect! @!input)
-      create-page? (page-handler/create! @!input {:redirect? true}))
-    (if create-class?
-      (state/pub-event! [:class/configure (db/entity [:block/name (util/page-name-sanity-lc class)])])
-      (state/close-modal!))))
+    (p/do!
```

Comment:
Use `p/do!` here because all the fns that call `db/transact!` are asynchronous now.